metals/src/main/scala/scala/meta/internal/metals/BuildServerConnection.scala

```diff
@@ -180,6 +180,20 @@ class BuildServerConnection private (
       .map(address => URI.create(address.getUri))
   }
 
+  def jvmRunEnvironment(
+      params: JvmRunEnvironmentParams
+  ): Future[JvmRunEnvironmentResult] = {
+    register(server => server.jvmRunEnvironment(params)).asScala
+      .map(env => env)
+  }
+
+  def jvmTestEnvironment(
+      params: JvmTestEnvironmentParams
+  ): Future[JvmTestEnvironmentResult] = {
+    register(server => server.jvmTestEnvironment(params)).asScala
+      .map(env => env)
+  }
+
```

Comment:
You should check if envinronment methods are supported by build server. While ago, I had sth along those lines. There is one question though, how client can know if environment endpoints are supported? 
1. Somehow inform client about that
2. Make result type optional so client can handle empty case and fallback to the DAP?
3. Client should try catch request for this method as it may fail now
```scala
  def jvmRunEnvironment(
      params: JvmRunEnvironmentParams
  ): Future[JvmRunEnvironmentResult] = {
    if (initialConnection.capabilities.getJvmRunEnvironmentProvider()) {
      register(server => server.jvmRunEnvironment(params)).asScala
    } else {
      scribe.warn(
        s"${initialConnection.displayName} does not support `buildTarget/jvmRunEnvironment`, unable to fetch run environment."
      )
      val empty = new JvmRunEnvironmentResult(Collections.emptyList)
      Future.successful(empty)
    }
  }

  def jvmTestEnvironment(
      params: JvmTestEnvironmentParams
  ): Future[JvmTestEnvironmentResult] = {
    if (initialConnection.capabilities.getJvmTestEnvironmentProvider()) {
      register(server => server.jvmTestEnvironment(params)).asScala
    } else {
      scribe.warn(
        s"${initialConnection.displayName} does not support `buildTarget/jvmTestEnvironment`, unable to fetch test environment."
      )
      val empty = new JvmTestEnvironmentResult(Collections.emptyList)
      Future.successful(empty)
    }
  }
```

Reply:
I used your suggestion, but for now using only jvmRunEnvironment since we don't really have a need yet for test.